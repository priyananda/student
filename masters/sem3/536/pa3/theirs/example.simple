// Note: this is not an exhaustive test, but is provided so that
// students can see an example of the desired result of Unparsing.

int    a; // a comment


int   aa;

bool   b;

//    a comment
void a(int p, String q, bool r) { // a comment
  int a; bool b;
  a = 0;
  if (a == -17) { return; }
  else {
    while (0 == -1) { a = 0; b = 9; }
  }
  return;
}

String s;

int x;

void a(int p, String q, bool r, float s) {
  if (b != !!!true) {
    a = false;
    s = 456.78;
    write (s*3 + "abd\"\\");
  }
  if (b && c || d) {
    a = b < c;
    a = b > c;
    a = b <= c;
    a = b >= c;
    a();
    a(x*y,y+z,-z);
  }
  return;
}
