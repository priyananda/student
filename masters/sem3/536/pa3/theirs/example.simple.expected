int a;
int aa;
bool b;

void a(int p, String q, bool r) {
    int a;
    bool b;

    a = 0;
    if ((a == -17)) {
        return;
    }
    else {
        while ((0 == -1)) {
            a = 0;
            b = 9;
        }
    }
    return;
}

String s;
int x;

void a(int p, String q, bool r, float s) {
    

    if ((b != !!!true)) {
        a = false;
        s = 456.78;
        write (((s * 3) + "abd\"\\"));
    }
    if (((b && c) || d)) {
        a = (b < c);
        a = (b > c);
        a = (b <= c);
        a = (b >= c);
        a();
        a((x * y), (y + z), -z);
    }
    return;
}
